### DISCUSSION TITLE
Shutter speed, aperture in the camera and focus for bokeh

### DISCUSSION THREAD

**Discussion by ihorpolynko** [mortuusars/Exposure#117] [Ideas]
Hello! Thanks for your work and mod! How does the shutter speed on the camera affect this particular mod? Will it be possible in the future to add manual focus to get bokeh in the background? It would be cool if the bokeh increased depending on the zoom range and aperture value

**Comment by mortuusars** [comment]
Shutter speed affects brightness of the image. 

Focusing (and all that's come from it - bokeh, for example), long exposures, motion blur (from long exposure) - most likely will not be implemented. 
It's really difficult to do, and even if possible - required too much effort and knowledge, that I don't have.

**Comment by tobi-blackscale** [comment]
I have an idea for motion blur and exposure that should work at least somewhat...
But it depends on the rate you can actually capture the screen and how efficiently you can modify the image with per pixel math.

Take a picture every x ticks while the shutter is open and divide the brightness info by a number (should be derived from your current math and the sample rate...). Then add the pictures together. if all pictures were the same or almost the same, they will add up to the single image, if something moved, it will be blurry. (this only works for long exposures, as shutter values above 20 are less than a tick anyways, for these your current math has to be used.)

The frames do not need saving, if you can do the math quick enough, but as rendering is done client side, dumping them to disk or ram and processing them later would also be an option.

**Reply by tobi-blackscale** [reply]
Second thought:
A similar process could also be used for making prints, switching to a photo paper and timed exposure process, allowing effects like blending images by exposing the same paper with multiple negatives for shorter times. 
This would also allow fixing brightness a bit by exposing the paper shorter or longer.
The chromatic process could also be moved into that by using colour filters. Or just colour tints by exposing the photo in white for some time, then switching to coloured light.
